target_chains/ethereum/contracts/forge-test/Pyth.Price.t.sol

```diff
@@ -0,0 +1,281 @@
+        assertEq(result.price, 5000, "Multiplication result should be 5000");
+        assertEq(result.expo, -16, "Exponents should be added");
+        assertEq(
+            result.publishTime,
+            1000,
+            "PublishTime should be the minimum of the two"
+        );
+    }
+
+    function testNormalize() public {
+        PythStructs.Price memory price = PythStructs.Price({
+            price: 2 * int64(PD_SCALE),
+            conf: 3 * PD_SCALE,
+            expo: 0,
+            publishTime: 1000
+        });
+        PythStructs.Price memory result = harness.normalize(price);
+        assertEq(
+            result.price,
+            (2 * int64(PD_SCALE)) / 100,
+            "Price should be normalized to 10^9"
+        );
+        assertEq(
+            result.conf,
+            (3 * PD_SCALE) / 100,
+            "Confidence should be normalized"
+        );
+        assertEq(result.expo, 2, "Exponent should be adjusted");
+    }
+
+    function testScaleToExponentUpscale() public {
+        PythStructs.Price memory price = PythStructs.Price({
+            price: 1000,
+            conf: 50,
+            expo: -8,
+            publishTime: 1000
+        });
+        PythStructs.Price memory result = harness.scaleToExponent(price, -6);
+        assertEq(result.price, 10, "Price should be scaled up");
+        assertEq(
+            result.conf,
+            0,
+            "Confidence should be scaled (rounded down to 0)"
+        );
+        assertEq(result.expo, -6, "Exponent should be adjusted to target");
+    }
+
+    function testToUnsignedPositive() public {
+        (uint64 unsignedValue, int64 sign) = harness.toUnsigned(100);
+        assertEq(unsignedValue, 100, "Unsigned value should be 100");
+        assertEq(sign, 1, "Sign should be positive");
+    }
+
+    function testToUnsignedNegative() public {
+        (uint64 unsignedValue, int64 sign) = harness.toUnsigned(-100);
+        assertEq(unsignedValue, 100, "Unsigned value should be 100");
+        assertEq(sign, -1, "Sign should be negative");
+    }
+
+    function testFraction() public {
+        PythStructs.Price memory result = harness.fraction(10, 5);
+        assertEq(result.price, 2 * 10 ** 9, "Fraction result should be 2");
+        assertEq(result.expo, -9, "Exponent should be -9");
+    }
+
+    function testAffineCombination() public {
+        PythStructs.Price memory y1 = PythStructs.Price({
+            price: 100,
+            conf: 10,
+            expo: -4,
+            publishTime: 0
+        });
+        PythStructs.Price memory y2 = PythStructs.Price({
+            price: 100,
+            conf: 10,
+            expo: -4,
+            publishTime: 0
+        });
+        PythStructs.Price memory result = harness.affineCombination(
+            0,
+            y1,
+            10,
+            y2,
+            5,
+            -9
+        );
+        assertEq(result.price, 10 ** 7, "Affine combination should be 150");

```

Comment:
this assert message is wrong. i think this test case is wrong too? should be 200 for y2, i think?